### PR TITLE
Fix #6606: Do not compute S on negative Ints

### LINKED ISSUE #6606: Compile-time successor of negative numbers
Should `S[-1]` work and if yes (as it currently is) how should we handle the following? (now it doesn't type check).

```scala
val t1: scala.compiletime.S[-1] = 0 // OK
val t2: scala.compiletime.S[Int] = 1 // OK
val t3: scala.compiletime.S[Int] = 0 // this is an error now
```

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
`S` is the Peano successor. By definition they only work on positive numbers.

I will add some documentation.

**Comment by nicolasstucki** [comment]
The only documentation we had so far was
https://github.com/lampepfl/dotty/pull/7544/files#diff-1b5c8ce1eaca91d793eef217487d5a59R1018-R1021. Which clearly states that they are natural number. Maybe it is worth considering having an actual alias for `Nat` with would be an intrinsinc definition for `type Nat = 0 | 1 | 2 | ... | 2147483647`.

File: tests/neg/i6606.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+val t0: Int = 0
+val t1: scala.compiletime.S[-1] = 0 // error
+val t2: scala.compiletime.S[Int] = 1
```
Comment (liufengyun) [inline tests/neg/i6606.scala]: It is not clear to me why this line type checks.

Comment (nicolasstucki) [reply tests/neg/i6606.scala]: I guess it is because type `1 =:= S[0] <:< S[Int]`

Comment (liufengyun) [reply tests/neg/i6606.scala]: We need to figure out the precise spec for `S`, whether it's co-variant, contra-variant, non-variant.

**Pull request merged**